demos/mqtt/mqtt_demo_lightweight/mqtt_demo_lightweight.c

```diff
@@ -781,7 +797,16 @@ static void mqttProcessIncomingPacket( NetworkContext_t * pNetworkContext,
          * packet. Since CONNACK is already processed, session present parameter is
          * to NULL */
         result = MQTT_DeserializeAck( &incomingPacket, &packetId, &sessionPresent );
-        assert( result == MQTTSuccess );
+
+        if( ( incomingPacket.type & 0xf0 ) == MQTT_PACKET_TYPE_SUBACK )
```

Comment:
A suback is always 0x90, so a mask isn't necessary
```suggestion
        if( incomingPacket.type == MQTT_PACKET_TYPE_SUBACK )
```